hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/cmd/UpdateCollectionIntervalsCommand.java

```diff
@@ -0,0 +1,200 @@
+import org.hawkular.bus.common.BasicMessageWithExtraData;
+import org.hawkular.bus.common.BinaryData;
+import org.hawkular.cmdgw.api.UpdateCollectionIntervalsRequest;
+import org.hawkular.cmdgw.api.UpdateCollectionIntervalsResponse;
+import org.hawkular.dmr.api.DmrApiException;
+import org.hawkular.dmr.api.OperationBuilder;
+import org.hawkular.dmr.api.OperationBuilder.CompositeOperationBuilder;
+import org.hawkular.dmr.api.OperationBuilder.OperationResult;
+import org.jboss.as.controller.PathAddress;
+import org.jboss.as.controller.client.ModelControllerClient;
+
+/**
+ * Update the specified metric and avail type collection intervals.  Performs a bulk update to prevent
+ * excessive agent restarts. Because metric types are not guaranteed to be consistent across agents, it is
+ * not a failure if a requested metric type does not exist.
+ *
+ * @author Jay Shaughnessy
+ */
+public class UpdateCollectionIntervalsCommand
+        extends AbstractResourcePathCommand<UpdateCollectionIntervalsRequest, UpdateCollectionIntervalsResponse> {
+    private static final MsgLogger log = AgentLoggers.getLogger(UpdateCollectionIntervalsCommand.class);
+    public static final Class<UpdateCollectionIntervalsRequest> REQUEST_CLASS = UpdateCollectionIntervalsRequest.class;
+
+    public UpdateCollectionIntervalsCommand() {
+        super("Update Collection Intervals", "DMR Nodes");
+    }
+
+    /** @see org.hawkular.agent.monitor.cmd.AbstractResourcePathCommand#createResponse() */
+    @Override
+    protected UpdateCollectionIntervalsResponse createResponse() {
+        return new UpdateCollectionIntervalsResponse();
+    }
+
+    @Override
+    protected BinaryData execute(
+            ModelControllerClient mcc,
+            EndpointService<DMRNodeLocation, DMRSession> endpointService,
+            String modelNodePath,
+            BasicMessageWithExtraData<UpdateCollectionIntervalsRequest> envelope,
+            UpdateCollectionIntervalsResponse response,
+            CommandContext context,
+            DMRSession dmrContext) throws Exception {
+
+        UpdateCollectionIntervalsRequest request = envelope.getBasicMessage();
+        Map<PathAddress, String> metricTypes = filterUpdates(mcc, request.getMetricTypes(), false);
+        Map<PathAddress, String> availTypes = filterUpdates(mcc, request.getAvailTypes(), true);
+
+        if (isEmpty(metricTypes) && isEmpty(availTypes)) {
+            log.debug("Skipping collection interval update, no valid type updates provided.");
+            return null;
+        }
+
+        CompositeOperationBuilder<?> composite = OperationBuilder.composite();
+        addUpdates(composite, metricTypes);
+        addUpdates(composite, availTypes);
+        // Don't let the restart hooks kick in, instead do a deferred restart
+        //composite.allowResourceServiceRestart();
+        composite.byNameOperation("start")
+                .address(PathAddress.parseCLIStyleAddress("/subsystem=hawkular-wildfly-agent/"))
```

Comment:
This won't work on EAP6, please to use this: [WildflyCompatibilityUtils.parseCLIStyleAddress](https://github.com/josejulio/hawkular-agent/blob/b873a144a7b2bf6c8e081ffcd583f23b55a1c515/hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/util/WildflyCompatibilityUtils.java#L119-L119)

`WildflyCompatibilityUtils.parseCLIStyleAddress("/subsystem=hawkular-wildfly-agent/")`